x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/topmetrics/InternalTopMetrics.java

```diff
@@ -95,9 +95,9 @@ public Object getProperty(List<String> path) {
         assert topMetrics.size() == 1 : "property paths should only resolve against top metrics with size == 1.";
         MetricValue metric = topMetrics.get(0).metricValues.get(index);
         if (metric == null) {
-            return Double.NaN;
+            return null;
         }
-        return metric.numberValue();
+        return metric.getValue().getKey();
```

Comment:
This'll return a `BytesRef` if you fetch from a `keyword` or `ip` or `bytes` field. Would it be more useful to you if I formatted it? Or do you want the raw bytes?

Reply:
In my local version I have a `valueAsString` method, I don't think I need this interface.

I will ping you on my PR once I have it ready.